Python/bytecodes.c

```diff
@@ -1425,6 +1422,10 @@ dummy_func(
             _PyInterpreterFrame *gen_frame = frame;
             frame = tstate->current_frame = frame->previous;
             gen_frame->previous = NULL;
+            FT_ATOMIC_STORE_INT8_RELEASE(gen->gi_frame_state, FRAME_SUSPENDED + oparg);
+#ifdef Py_GIL_DISABLED
+            ((_PyThreadStateImpl *)tstate)->gen_last_frame_state = FRAME_SUSPENDED + oparg;
+#endif
```

Comment:
Would it make sense to factor this out into a helper function? It's duplicated in `ceval.c`. Also, having helper to update both would reduce the chance that we update one without the other.

Reply:
I've refactored it out. I also moved the functions to `ceval_macros.h` from `pycore_genobject.h` because:

1) they are only used in ceval.c and bytecodes.c
2) I want to avoid adding additional includes to `pycore_genobject.h` because I think it's pulled in by internal headers that some third-party projects tend to include.